pkg/util/log/log.go

```diff
@@ -452,99 +448,155 @@ func formatErrorc(message string, context ...interface{}) error {
 // scrubAndLogFunc, and treating the variadic args as the message.
 func log(logLevel seelog.LogLevel, bufferFunc func(), scrubAndLogFunc func(string), v ...interface{}) {
 	l := logger.Load()
-	if l != nil && l.inner != nil && l.shouldLog(logLevel) {
+
+	if l == nil {
+		addLogToBuffer(bufferFunc)
+		return
+	}
+
+	l.l.Lock() //// If we change the logic here we will want to change in addLogToBuffer too
+	defer l.l.Unlock()
+
+	if l.inner != nil && l.shouldLog(logLevel) {
 		s := BuildLogEntry(v...)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏`BuildLogEntry` can be somewhat costly, that's why it was done before locking, not sure how but we might want to keep the old behavior

Reply:
Maybe just compute it before locking, even if we might not use it if the logger hasn't been initialized yet, I think we can optimize for the most common code path (logger initialized)